packages/core/lib/commands/publish.js

```diff
@@ -6,16 +6,21 @@ var command = {
     usage: "truffle publish",
     options: []
   },
-  run: function(options, done) {
+  run: async function (options, done) {
```

Comment:
Oh I should note that it's not wise to make a function like this be `async` and take a callback. I'd suggest instead using `callbackify` (`const { callbackify } = require("util")`) and not bothering with the callback arg at all. See [Node `callbackify` docs](https://nodejs.org/api/util.html#util_util_callbackify_original)